### ISSUE TITLE
Rise and Fall and 1.0.0.216

### ISSUE THREAD

**Issue by Azurency** [Azurency/CQUI_Community-Edition#283]
The first expansion is here. Rise and Fall looks great and of course as it adds news things we have to work on the mod to make it compatible.

We'll keep the good practice to use this issue as the "master issue" of all the Rise and Fall patch related issue. ⚠️ **So please report your bugs here and not on separate issues**.

#### 👷‍♀️ About patching CQUI

Yes, we're aware of the issues. It's almost always resolved by merging the patch changes (and there're were some, in this 1go patch) into the mod files, but often require us to adapt the mod changes to new naming, functions, code removed/moved, etc.
That's why it may take some time (I mean more than a day, waaay more), if the patch is big or if there're some breaking changes.

#### 🤔 Rise and Fall and Vanilla patch 1.0.0.216
I don't know how it'll turn out, I'll edit this part later for more precisions. For now I think that we'll do something like that : 

1. Update cqui for the 1.0.0.216 patch **without the expansion**
2. Move on the **Rise and Fall expansion** and focus on it for any future update

I'm new to modding for Civilization and don't know how expansions are dealt with, so I prefer to proceed this way (sorry for those with the expansion that might have to wait a little more).

#### 🔨 How we'll proceed

We'll try to (roughly) follow these steps, maybe not in that exact order : 
- [ ] Make the UI files diff of the vanilla patch and have a quick look at them
- [ ] Make a list of which files in the mod need to be looked at for vanilla
- [ ] Create a new `Patch-1.0.0.216` branch
- [ ] Patch the changes for CQUI _(the hard work)_
- [ ] Look at the integrations of CQUI, maybe they already have patched on their end
- [ ] Patch the changes for the integrations _(also not easy)_
- [ ] Test the changes
- [ ] Make the UI files diff of the Rise and Fall expansion and have a quick look at them
- [ ] Make a list of which files in the mod need to be looked at for Rise and Fall
- [ ] Create a new `Rise-and-Fall` branch
- [ ] Patch the changes for CQUI and its integrations _(the hard work)_
- [ ] Test the changes
- [ ] Merge the `Rise-and-Fall` back into master
- [ ] Address the non-resolved issues (if any) that still exist after the patch and that may require more work

#### TL;DR
- Do not create new issue for the new Rise and Fall patch
- Post your bugs here
- We'll update CQUI  for the base game first then for the expansion
- We're working on it 😉

**Comment by Azurency** [comment]
I made the diff for the base game patch, they're here : https://github.com/Azurency/Civ6-UIFiles/commit/6c8d97f59092f12d35d92fe645ad4f8673dd5ffe
It's a big patch, **160 changed files** (+5771, -3893) it's less than the Fall and Summer 2017 patches. But still not nothing.

Now that we have the diffs, I'll soon make a list of which files in the mod need to be looked at and changed (and how much it changed, which generally means how much work needs to be done).

**Comment by wayneb64** [comment]
Awesome news this is being worked on! Is there any thought to keeping a version separate for Vanilla and not merging once the R&F version is ready? I certainly plan on upgrading to R&F and maybe everyone does so this would be pointless. The real question is how painful would it be to have separate versions of the mod on Steam? If it would just cause more problems, than just forget the idea.

**Comment by Azurency** [comment]
Here's a list of files that needs to be changed : 

#### CQUI changes

- [ ] `actionpanel.lua`, few changes - _easy_
- [ ] `choosers/civicschooser.lua`, few changes - _easy_
- [ ] `choosers/researchchooser.lua`, few lines added - _easy_
- [ ] `civ6common.lua`, few changes - _easy_
- [ ] `civilopedia/civilopediasupport.lua`, new additions - _require some work_
- [ ] `diplomacyactionview.lua`, lot of changes (UI related, gossip, ...) - _require work_
- [ ] `diplomacyactionview.xml`, some changes - _require work_
- [ ] `ingame.lua`, 6-line change - _easy_
- [ ] `launchbar.lua`, some new things - _require some work_
- [ ] `panels/citypanel.lua`, 7 new lines - _easy_
- [ ] `panels/citypanel.xml`, very few changes - _easy_
- [ ] `panels/citypaneloverview.lua`, lots of changes (with an `-- HACK:` comment that may be related to #228) - _require work_
- [ ] `panels/citypaneloverview.xml`, very few changes - _easy_
- [ ] `panels/notificationpanel.lua`, few changes - _easy_
- [ ] `panels/productionpanel.lua`, some changes (love their comment on how the code is messy and needs to be refactored haha) - _require work_
- [ ] `panels/unitpanel.lua`, lots of changes - _require work_
- [ ] `panels/unitpanel.xml`, some changes - _require some work_
- [ ] `partialscreenhooks.lua`, some changes - _require some work_
- [ ] `partialscreenhooks.xml`, few changes - _easy_
- [ ] `partialscreens/citystates.lua`, some new function - _easy_
- [ ] `popups/boostunlockedpopup.lua`, few changes - _easy_
- [ ] `popups/greatpeoplepopup.lua`, few changes (interesting new possibilities to queue/dequeue a popup) - _easy_
- [ ] `popups/mappinpopup.lua`, some changes - _require some work_
- [ ] `popups/mappinpopup.xml`, 1 change - _easy_
- [ ] `popups/techciviccompletedpopup.lua`, few changes - _easy_
- [ ] `popups/wonderbuiltpopup.lua`, very few changes - _easy_
- [ ] `productionpanel.xml`, very few changes - _easy_
- [ ] `screens/civicstree.lua`, some changes - _require some work_
- [ ] `screens/governmentscreen.lua`, LOTS of changes - _require work_
- [ ] `screens/governmentscreen.xml`, lots of changes - _require work_
- [ ] `screens/techtree.lua`, few changes - _easy_
- [ ] `strategicview_mapplacement.lua`, few changes - _easy_
- [ ] `techandcivicsupport.lua`, some changes - _require some work_
- [ ] `tooltips/plottooltip.lua`, few changes - _easy_
- [ ] `toppanel.lua`, few changes - _easy_
- [ ] `toppanel.xml`, few changes - _easy_
- [ ] `unitflagmanager.lua`, small changes - _easy_
- [ ] `worldinput.lua`,  some changes - _require some work_
- [ ] `worldtracker.lua`, few changes - _easy_
- [ ] `worldtracker.xml`, very few changes - _easy_
- [ ] `worldview/citybannermanager.lua`,  some changes - _require some work_
- [ ] `worldview/citybannermanager.xml`, few changes - _easy_
- [ ] `worldviewiconsmanager.lua`, some additions - _should be easy_

#### Integrations changes

BES :
- [ ] `choosers/espionagechooser.lua`, very few changes - _easy_
- [ ] `partialscreens/espionageoverview.lua`, some changes - _require some work_
- [ ] `partialscreens/espionageoverview.xml`, 1 change - _easy_

BTS : 
- [ ] `choosers/traderoutechooser.lua`, some changes - _require work_
- [ ] `choosers/traderoutechooser.xml`, some changes - _require work_
- [ ] `tradeoverview.lua`, few changes - _easy_
- [ ] `tradeoverview.xml`,  some changes - _require some work_

IDS : 
- [ ] `diplomacydealview.lua`, some changes - _require work_
- [ ] `diplomacydealview.xml`, few changes - _easy_

ML : 
- [ ] `minimappanel.lua`, few changes - _easy_
- [ ] `panels/modallenspanel.xml`, 1 change - _easy_

That's a lot of small changes everywhere, and some big changes. Might take about the same amount of time that it took us to fix the mod after the fall 2017 patch. At the very least one week.

**Comment by Azurency** [comment]
@wayneb64 I don't know if we'll do two versions on steam, we'll have to think about it.
Whatever we decide, there will be a dedicated branch here on GitHub and a link to manually download the CQUI version compatible with the base game.

**Comment by astog** [comment]
I guess it is time to break my hibernation huh. I'll start patching stuff right away. No promises on time though.

**Comment by oerms** [comment]
I will continue working on the _easy_ files tomorrow or on Monday evening. To get some checkmarks done. Always good psychologically. ;)

**Comment by eporres** [comment]
Looking to contribute however I can, and I just have one question. What's the best way to avoid duplicate work? Is checking the pull requests the only way to get a feeling for what other people are doing?

**Comment by Azurency** [comment]
@eporres Well... There should be a way. But right now yes, checking the pull request + the files that are checked here in the issue as well as taking a look at the commit on the patch branch is the way to know what other people work on.
Maybe I should create a new Chanel in the Discord (https://discord.gg/RAFEpAK) so anyone can tell on which file they're working.

**Comment by eporres** [comment]
@Azurency Yeah, I think that's a good idea. It might even prove beneficial in the future as a place for discussion/feedback.

**Comment by ihendriks** [comment]
@jeffpierce I think a kanban for all these small diff files will not be worth your time. As long as @Azurency keeps the checklist above in check it should be alright.

**Comment by bolbass** [comment]
Hi, @Azurency. I'm trying to update real housing function but I have some troubles with it. It seems like R&F expansion doesn't use 'citybannermanager.lua' at all. Any changes in this file have no effect. When I even erase all the lines banners still exist. Am I wrong or what do you think of it?

**Comment by Azurency** [comment]
@bolbass We're not working on Rise and Fall yet, but on the base game patch. I still haven't look at R&F and did not even play a game. That what I thought, the expansion looks like it's loaded "like a mod" with its files replacing the base game files.

**Comment by rojona** [comment]
@Azurency Are the R&F changes inactive when choosing another rule set? Does that mean that the base game patch version of CQUI could be installed over a game with R&F but be played as if it's vanilla?

**Comment by BlackSmokeDMax** [comment]
Hmm, hadn't considered the whole vanilla and expansion thing. Wonder how much the two are different with:

1. R&F not selected in Steam (in Library - DLC section of the game)
2. R&F not selected in mods menu
3. or with just not using the R&F ruleset via the game setup dialog

Sounds like this could get very messy.

**Comment by oerms** [comment]
The base game files should be done. Now it's about testing and finding what CQUI does not work anymore because of game changes.
Does anybody know which integrations are already being worked on? Any word from the respective owners?

**Comment by rojona** [comment]
@oerms Are those files available as a test release?

**Comment by oerms** [comment]
They are not merged yet into Azurencys repo...
But you can manually pull all the forks and try them out.

**Comment by Azurency** [comment]
@oerms well @astog seems to have updated something on BES, but we might not wait for him to update its other mods (BTS, ML).
IDS will need to be updated by us as no one is providing any more support (I updated it for the last summer and fall patch, it's quite an heavily modified file compared to the base game one)

**Comment by Azurency** [comment]
**A small progress update** : I've done the review of all the pull requests on GitHub, waiting for some changes by the contributors but it looks great. 
We'll then move on to the integrations. I'll update IDS myself as it's not maintained. We'll pull the changes made by @astog on BES. For his other mods, we'll see this weekend if it's updated or not, if it's not will start working on it (and then post a pull request on his repo so he can update them easily)

**Comment by Azurency** [comment]
I think we can start testing now, by downloading the `Patch-1.0.0.216` branch here. But keep in mind that espionnage, trade route, lenses and deal view are not yet updated. If you find a bug somewhere else, you can report it here.

**Comment by eporres** [comment]
@Azurency Anything specific you want us to do in order to keep the issues we find more organized?

Just from starting a quick game, I see the following:

- **Delayed Eureka popups**
Eureka for farming a resource did not show up until I placed a district down even though the farm was built several actions before (on the same turn).

- **Unable to remove individual items from the production queue**
Right clicking an item in the production queue does not remove it. Fairly certain this was a feature before, but please correct me if I'm mistaken.

**Comment by Bruellhusten** [comment]
> Delayed Eureka popups
> Eureka for farming a resource did not show up until I placed a district down even though the farm was built several actions before (on the same turn).

This is broken without CQUI as well. 
So more a goodie if the mod fixes this as well.

**Comment by wayneb64** [comment]
I have been playing Vanilla with the 216 patch and I have not seen this issue with the Eureka popups. Is this maybe an R&F issue?

**Comment by Jim102469** [comment]
Did the 220 patch on Feb 13 mess up anything that you had already worked on? I would love to help test if you like. It's what I do for a living as a software test engineer. I will pull down the branch and take a look.

**Comment by BlackSmokeDMax** [comment]
@Bruellhusten Haven't seen that behavior, either before or after the Feb 13th patch.

I am playing with:
R&F
All DLC
following mods:
Sukritacts: "civ selection screen" and "improved mod screen"
Caps Lock
Improved Kineticam

**Comment by Bruellhusten** [comment]
I just started a new game and tried to reproduce it, but it worked. 
When I wrote this last night I was thinking of several other eurekas that popped up late. Like the "get 3 archers" for machinery. It was in a multiplayer session though. But it happened more than once and explicitly after another action. Like you fulfill the archer requirement by upgrading another slinger. And the eureka notification shows up 3 seconds later when you do the next action, like placing a new district.
I only saw this behavior after R&F release.

I will try to reproduce this. 

Am running on the latest patch and R&F as well. But just the Australia DLC. 
Mods: 
Yet (not) another Maps Pack

**Comment by wayneb64** [comment]
There is a minor but annoying bug I noticed after the latest mod drop to Steam that caused the citizen highlights to get 'confused' in the city view. While in map view with the citizen highlights (not sure what to call it) enabled you can mouse over a city header and it would show the citizens with either green or yellow borders on their hex to indicate locked or not. The bug was when you actually clicked on the city header and opened the city it would switch states and you could NOT see these highlights unless you clicked a citizen again. Easy to work around, but annoying in that it always happens unless you click a different city header while already in city view. It's just a state change bug somewhere but would be nice of that where fixed with the rest of the issues caused by the patch.

**Comment by wayneb64** [comment]
I see they patched the base game again, did that mess things up again with the mod updates?

**Comment by alexeyOnGitHub** [comment]
would it be possible to release a "beta" version of CQUI for RiseAndFall with the most basic features first, before releasing a "full" version (with everything else also fixed)?

e.g. with there basic features - 
* production queue
* city working tiles

side note / off-topic - separating this mod into several smaller independent modules (with their own release schedules) would hopefully enable faster releases in the future after Civ6 updates or expansion packs.

**Comment by ProustIdee** [comment]
You can use the 1.0.0.216 branch https://github.com/Azurency/CQUI_Community-Edition/archive/Patch-1.0.0.216.zip

Unzip into \Documents\My Games\Sid Meier's Civilization VI\Mods\

**Comment by alexeyOnGitHub** [comment]
@ProustIdee  I unpacked that file to "Mods" folder as you suggested. started Civ6 RaF. I see two "CQUI" mods on "additional content" page - one disabled (from Steam workshop) and another one with the same name - enabled (I assume this is the one from the Zip file). 
then I resumed an existing game. 
cannot move units with a mouse. maybe the old mod is still loaded somehow?

**Comment by Azurency** [comment]
@wayneb64 fixed already in def7bba82e166689cb744b82fb39c3998576407c (related to #228)
@eporres thx for reporting that, I'll fix the production queue and will take a look at the envoys. 

**About the integrations :**
We have to start working on them now, **BTS and ML** need to be upgraded if anyone wants to contribute you can merge the changes and create a pull request.
I'll pull the changes of BES from @astog repo, he updated it recently.

**Comment by Azurency** [comment]
@alexeyOnGitHub The mod is not compatible with Rise and Fall yet, right now we're working on updating it for the latest base game patch (without Rise and Fall expansion enabled).

**Comment by eporres** [comment]
@Azurency For the integrations, do you guys usually contribute with a pull request to their respective repos and then bring the changes over to CQUI as a patch? I guess I don't fully understand what you mean by merge and then open a pull request. Merge in their latest changes?

**Comment by Azurency** [comment]
@eporres I mean, do exactly like the cqui files in fact. Merge the changed of the patch to the files and open a pull request here, then we'll send a pull request to the respective repos.

**Comment by eporres** [comment]
@Azurency The only reason I asked was because @astog seems to have already begun fixing stuff for this latest patch. At least it seems that way, both repos have a branch for the patch and are ahead of master.

**Comment by Azurency** [comment]
@eporres yes, we should bring his changes and test it/complete it if it's not finished

**Comment by Azurency** [comment]
@eporres regarding the "Quick suzerain indicator" what indicator are you referring to ? The city states report screen on the panel ?

**Comment by astog** [comment]
I hate to give timelines, since people then get expectations 😄 , but I should be finishing my local patches tonight. Hopefully port them into CQUI by tomorrow.

**Comment by eporres** [comment]
@Azurency I'm starting to wonder if I'm simply not understanding what this number in the UI means. Either way, I'm attaching a couple of images so that this can be clarified (at least for me 😛). The first image shows Victoria as the suzerain for Carthage with four envoys. However, if I bring up the panel with the number of envoys from each civilization, Victoria has actually only sent three envoys to Carthage. Does the four in this case not represent the number of envoys that the suzerain has sent?

![image](https://user-images.githubusercontent.com/22130077/36347221-d1e535b4-141f-11e8-83dd-68c2e1aac879.png)

![image](https://user-images.githubusercontent.com/22130077/36347243-45f63610-1420-11e8-86af-0c3f6262c4c7.png)

**Comment by BlackSmokeDMax** [comment]
@eporres the number 4 in that case is the quantity *you* need to become suzerain. So in that column right above/below each other it shows 3 (your current) and 4 (# required for you to become suzerain. For Carthage you have 1 and you would need to get to 4 to become suzerain.

**Comment by eporres** [comment]
@BlackSmokeDMax Thanks for the explanation. Now I feel dumb.
@Azurency Never mind what I said. Apparently I need to get more familiar with this mod 😄.

**Comment by gasmasher** [comment]
@Bruellhusten I had the delayed eureka in a game with all DLC enabled tonight. When I upgraded a Sumerian War Cart all the eureka dialog triggered. I am using the Azurency patch (zip) from 2/17/2018. The only other community mod I have is Detailed Worlds latest.

**Comment by MauiSven** [comment]
@Azurency 
Are the changes due to the 1.0.0.216-patch compatible with the 1.0.0.220-version?

**Comment by Azurency** [comment]
@MauiSven yes, the changes of 1.0.0.220 are here https://github.com/Azurency/Civ6-UIFiles/commit/a59c1d519ea282e15733fefa00e7a6e047e69634, and there's nothing

**Comment by deiga** [comment]
@gasmasher The delayed Eureka is a bug in the actual game and not in this mod.

**Comment by rasputinoz** [comment]
Downlaoded the patch to try and help test but it wont let me move the units at all.

**Comment by ihendriks** [comment]
> Downlaoded the patch to try and help test but it wont let me move the units at all.

@rasputinoz Did you download the right branch? The correct zip is [here](https://github.com/Azurency/CQUI_Community-Edition/archive/Patch-1.0.0.216.zip). And the mod doesn't work for Rise & Fall yet, so you want to make sure you're playing on standard ruleset, not R&F.

**Comment by wayneb64** [comment]
I tried the patch this weekend and got the same results. I did happen to still have the Steam version present but disabled. I had to delete BOTH, and restart, before I could the game working again. Maybe I should have removed the Steam version first, but that was not indicated anywhere.

**Comment by gasmasher** [comment]
Thanks @deiga, I get it.

Thanks to the folks working on this. It is working well enough for me to do some quick games.

**Comment by ghengiseanie** [comment]
BES and BTS have released R&F compatibles on Civ Downloads.. IF we drag them into CQUI folders/..? Is it that simple?

**Comment by rasputinoz** [comment]
excellent, the difference I unsubscribed from the Steam version and seems to work now, 

will play and take note of any bugs that havent already been seen

**Comment by astog** [comment]
@ghengiseanie Not exactly. The issue with the expansion they replace many of the UI scripts. So my mod in turn replaces their own replacement (I know, wtf). With CQUI I have to edit the modinfo to make it work. That is why it is taking longer then expected. I should have them done sometime this week (college is way too time consuming).

**Comment by camk16** [comment]
Just wanted to thank everyone for all the hard work you've been doing!

**Comment by Kree6427** [comment]
-As soon as I start the game, a massive frame drop takes place.
-When I settle a city, the city doesn't display a name, population number, etc over the settlement.
-I cannot open production at all and there is no way to open it, so I can't go beyond the first turn.
-When I want to move my only warrior unit, I can't move it at all. The route is stuck to some space off the screen and I can't move it.
-When I disable the mod it seems to infect my game files and causes me to have low framerate, an unusable production button, and the disability to not be able to move my units or advance the turn.
-The only solution I found was to verify the game cache via steam.
-BTW I only run the DirectX12 version of the game.
-Overall, I can't use this mod and I will keep it away from my mod list until a fix is made.

**Comment by rasputinoz** [comment]
Did you download the latest version fro GitHub works very well for me , only two issues so far
1. Icons from notifications remain on right hand side even after tight clicking on them and hang around for many turns for example promotion icon even after promoting unit
2. The Escape button isn’t working to load menu for saving etc

Other than that going well so far

Sent from my iPhone

On 24 Feb 2018, at 11:03 am, Kree6427 <notifications@github.com<mailto:notifications@github.com>> wrote:


-As soon as I start the game, a massive frame drop takes place.
-When I settle a city, the city doesn't display a name, population number, etc over the settlement.
-I cannot open production at all and there is no way to open it, so I can't go beyond the first turn.
-When I want to move my only warrior unit, I can't move it at all. The route is stuck to some space off the screen and I can't move it.
-When I disable the mod it seems to infect my game files and causes me to have low framerate, an unusable production button, and the disability to not be able to move my units or advance the turn.
-The only solution I found was to verify the game cache via steam.
-BTW I only run the DirectX12 version of the game.
-Overall, I can't use this mod and I will keep it away from my mod list until a fix is made.

**Comment by wayneb64** [comment]
I used to see the icons not being clearable now and then, long before the 216 patch. The GitHub version was unplayable for me and another player, not sure why but I suspect maybe the Steam version was confusing the GitHub version somehow as they both had identical folder names but no instructions where given about clearing out the Steam version being a requirement.

**Comment by Azurency** [comment]
I merged More Lenses today @astog with the latest changes on your repo. I still need to look at how to move the citizen management lens outside of the `minimappanel.lua` (with your new way to load lenses).

**Comment by astog** [comment]
Thanks for that @Azurency . I will have to use custom LuaEvents to do that, but my intention was to remove the older Government Lens like citizen management area that was really slow late game. Rather than that I will use the already working and faster Appeal Lens like highlighting.  As for BTS and BES they seem to be working so I should send you a pull request soon.

**Comment by Azurency** [comment]
@astog That would be great if you could do that, the government lens was not ideal. I merged BTS and BES thank you for updating it 😉

**Comment by Azurency** [comment]
**Update** : Everything is merged for the base game patch 1.0.0.216, now we'll have to test and fix the remaining issues (some of you already reported some, I'll take a look at that now).
To help the testing you can download the code in the branch `Patch-1.0.0.216`.

**Comment by the-gasman** [comment]
I have play tested for a couple of hours and not come across any bugs or unusual behaviour. Civ 220.

**Comment by MauiSven** [comment]
@thyeestes Where did you download that patch??

**Comment by the-gasman** [comment]
Go to the code tab, select branch: Patch-1.0.0.216, hit the green download button.

**Comment by thyeestes** [comment]
I downloaded the patch from here when I extract the zip the folder name is patch-1.0.0.216

**Comment by the-gasman** [comment]
That's the one I installed and everything seems to be working.

**Comment by rasputinoz** [comment]
issues: 
not displaying the other civs along top of window like it used to . no information at all
Escape key not working, have to go to top right of screen and click the Menu Button
When selecting a Worker it doesnt highlight the hexes that are recommnended to be worked on

**Comment by mustang05tim** [comment]
> issues:
not displaying the other civs along top of window like it used to . no information at all
Escape key not working, have to go to top right of screen and click the Menu Button
When selecting a Worker it doesnt highlight the hexes that are recommnended to be worked on

I am running the mod via a manual install, without the expansion, and I don't see any of those issues  yet.  I'm on turn 42 so far and I haven't had any issues that I have noticed so far.

**Comment by rasputinoz** [comment]
I will try deleting the mod then reinstalling the download again

Sent from my iPhone

On 26 Feb 2018, at 2:40 pm, mustang05tim <notifications@github.com<mailto:notifications@github.com>> wrote:


issues:
not displaying the other civs along top of window like it used to . no information at all
Escape key not working, have to go to top right of screen and click the Menu Button
When selecting a Worker it doesnt highlight the hexes that are recommnended to be worked on

I am running the mod via a manual install, without the expansion, and I don't see any of those issues yet. I'm on turn 42 so far and I haven't had any issues that I have noticed so far.

**Comment by rasputinoz** [comment]
![capture](https://user-images.githubusercontent.com/30841016/36653812-e07f5944-1b07-11e8-89d9-aeac058c4b6e.JPG)
refer to picture to see what i mean

**Comment by mustang05tim** [comment]
I think that I spoke too soon.  I kept playing until turn 47, and as soon as I researched Mysticism and it started to speak the "Mysticism, I like to say", it kept saying that on a loop and locked the game up completely.  Only by signing out of windows and signing back in was I able to get control back.  I tried it a second time, with the last auto saved file before the crash, and it crashed it again.  The ONLY mod except for the official DLC's (not Rise and Fall) installed is the 1.0.0.216 CQUI from there.   Attached is my config and save file. 

[autosave47.zip](https://github.com/Azurency/CQUI_Community-Edition/files/1757266/autosave47.zip)

**Comment by CptSpoon** [comment]
Same issue than mustang05tim for me.

**Comment by Xuampas** [comment]
Same problem when I found a religion. Funny loop tho.

**Comment by trafenrest** [comment]
I had the same problem and issued a bug report (its linke above)
if you like to continue with your game, de-activate the mod, load the saved game and play one turn, save the game. 
Then quit and activate CQUI again and you can continue.

**Comment by Azurency** [comment]
@Bruellhusten I fixed the eureka issue.

@mustang05tim @CptSpoon @trafenrest I can't reproduce the infinite loop, everything is fine for me, even when I load your save it's working fine. Can you post the logs of when you load your game and have the issue ? it's in the `Logs` folder next to the Saves and Mods folders.
@Xuampas Also not having infinite loop when I found a religion

@rasputinoz Can you post a save game and the logs of when you load the game and have the issue ? It would help.

**Comment by trafenrest** [comment]
[Logs.zip](https://github.com/Azurency/CQUI_Community-Edition/files/1760302/Logs.zip)
Here are the logs when it crashes.

**Comment by Azurency** [comment]
@trafenrest Are you sure you're using the latest version of CQUI from the patch branch ? I don't see the correct loaded dependency in the `modding.lua` for CQUI (it was updated lately). Also I just added the version number of CQUI to a print to help debugging. Can you download the latest version and repost your log with the issue please ?

**Comment by trafenrest** [comment]
As I said in the bug report and in my comment above, when I disable CQUI I can finish the round and save it. After that, I can enable CQUI and continue to play.

In which logfile do I see the version number now?
I just downloaded the current version and (hope I) activated it. Now when I load my saved game and try to move the prophet to the nearby holy place (or any othe of my units), it warns me that this will start a war with Japan - but I am Japan! Even if I accept to go to war, the prophet does not move.. so I cant found the religion and it does not crash

**Comment by trafenrest** [comment]
Sorry, I misunderstood you.
I will disable the other mods and try again

**Comment by Azurency** [comment]
@trafenrest in `Lua.log`. That's weird... you're on windows, base game (no Rise and Fall) update 1.0.0.216 ?

**Comment by trafenrest** [comment]
I'm on windows, base game, latest patch 1.0.0.**220** (334528). 

**Only CQUI and offical mods activated:**

Select Irenaeus, the great prophet and try to move him to the holy area near Kyoto. I'm ask if I want to create a war with myself-

**Comment by rasputinoz** [comment]
I not sure what has changed but started a new game today and it all looks like it working so loaded the game that had the issues but they are not there anymore. 
Will continue playing and see if the problem arises again.

Thanks for this amazing MOD you guys are amazing

**Comment by Azurency** [comment]
@trafenrest Logs and save game please 😉 
@rasputinoz I only fixed the eureka boost today, maybe it was something related to wrongly queued popups that caused some weird things to happens

Also to anyone reporting an issue : please post the save game and your logs, so I can reproduce/understand the issue faster

**Comment by trafenrest** [comment]
Here's the new log. The saved game is still the same, I included i for your convenience.
[HOJO TOKIMUNE 104 300 v. Chr.CRASH.Civ6Save.zip](https://github.com/Azurency/CQUI_Community-Edition/files/1760522/HOJO.TOKIMUNE.104.300.v.Chr.CRASH.Civ6Save.zip)

I modified the description in CQUI.modinfo to be sure to have the right one (local, not the steam) active.
Strange: I can't find a version number for CQUI in loa.log, but maybe I miss it. Also it says that there are still more mods active than I activated. My guess is that they are needed for the saved game?
[Logs.zip](https://github.com/Azurency/CQUI_Community-Edition/files/1760521/Logs.zip)

I quit CIV and started new, de-activated CQUI and loaded the save game again. Here are the logs of that run. I could use the prophet and create a religion
[Logs no CQUI.zip](https://github.com/Azurency/CQUI_Community-Edition/files/1760541/Logs.no.CQUI.zip)

**Comment by mustang05tim** [comment]
I will test again tonight when I get home.  Azurency's modification to the eureka boost (in Assets/UI/Popups/boostunlockedpopup.lua)  probably will fix my problem.  I'll test and capture logs from my save before and after the fix..  Maybe a diff of the logs will confirm this was the issue.

**Comment by Azurency** [comment]
@trafenrest You're still loading CQUI from steam... that's the issue. It's not the latest version. `workshop\content\289070\1125169247`

**Comment by trafenrest** [comment]
You mean I played with the steam version all the time? Beside that crash, I had NO issue with the steam version.
Even when I deactivate it? How stubborn! What can I do? Seems that (de-)activating is of no use for saved games..

**Comment by rasputinoz** [comment]
i had to delete the steam mod to ensure the download from here works

**Comment by gipfeli** [comment]
@trafenrest Go to My Documents/My Games/Sid Meier9s Civ. VI/ Mods and check if anything left over. Delete them all (+ Unsubcribe from Steam workshop), and manual reinstall with the test version from here.

**Comment by mustang05tim** [comment]
I tried again with the latest build (today's), and captured before and after logs.  I still am crashing.  Next I will try reverting to a few turns before the crash to see if that makes a difference as well.  

[InitialLogs.zip](https://github.com/Azurency/CQUI_Community-Edition/files/1761166/InitialLogs.zip)

[LogsAfterChange.zip](https://github.com/Azurency/CQUI_Community-Edition/files/1761167/LogsAfterChange.zip)

**Comment by mustang05tim** [comment]
Went a few turns back to a save a couple of moves before the endless speech loop, and still it is crashing when the mysticism speech occurs.   Atttached are my saves ( it crashes on turn 47) 

[autosave45and47.zip](https://github.com/Azurency/CQUI_Community-Edition/files/1761209/autosave45and47.zip)

**Comment by the-gasman** [comment]
Unfortunately I've just experienced the Mysticism bug too. Exactly as described, crash with looping audio.

**Comment by the-gasman** [comment]
Deleting CQUI allows mysticism to be discovered without problem (from the autosave of the same turn). CQUI can then be reinstalled from the next turn and the game continues normally.

**Comment by mustang05tim** [comment]
Will a future civics discovery that triggers a speech several turns down the road also cause the crash and audio loop?   I haven't got that far yet.  I also want to test it tonight with just the CQUI disabled, not uninstalled.

**Comment by mustang05tim** [comment]
Was able to get around it via simply disabling the mod, then loading the save that was crashing before, continue to the next turn, and then re-enable the mod.  I'm about 10 turns past that now, with more speeches loading, and haven't had the issue return (yet).  One thing I did notice, was that on my original save that was crashing, there were actually 2 different speeches that were going to happen, because both my research and my civic advanced on that turn.  Maybe that's the trigger?

**Comment by Sabrejack** [comment]
I did get a crash late game, I don't recall exactly which civic/tech it was, but the audio looped as described here.  

I have also had the mysticism crash, for me it seemed connected to the fact that I was boosted into finishing mysticism by founding a pantheon.  I was able to avoid the crash without disabling CQUI by right-click-dismissing the "found a pantheon" button for a few rounds until I had discovered Mysticism manually without boost.  Then I founded my pantheon, and everything was fine.  

Hope this helps.

**Comment by the-gasman** [comment]
The Eureka pop-up for archery came fully 6 turns after I killed a unit with a slinger. In fact, the eureka pop-up came when I completed researching archery.

**Comment by Rahghnall** [comment]
Hi!

!!Big thanks for the mod!!

bugs:
- "Show yield" randomly switching off, can't see what is causing
- Exclamation mark for units, sometimes disappering and reapper after i use the city screen
- the attached picture marked with red circle a menu. dont know what it is but clearly not working.

Don't own R&F, the mods i used with the CQUI:
Raging Barbarians
Gold Resource
Strategic Forts
Better Civilization Icons
Better Coastal Cities and Water Tiles
8 Ages Of Pace
Good Goody Huts
Unique District Icons
R.E.D. Modpack
Yet (not) Another Maps Pack
No Barbarian XP Cap
World Wonders for Everyone

Attaching save file as well
[GORGO 837 1928 AD.zip](https://github.com/Azurency/CQUI_Community-Edition/files/1766672/GORGO.837.1928.AD.zip)
the mysterious menu:
![menu](https://user-images.githubusercontent.com/36918552/36782741-2218a00e-1c7a-11e8-8405-471f78edcbbb.jpg)

Keep up the good work!

**Comment by radiation** [comment]
I'm using the newest patch from the 1.0.0.216 branch and having some issues that others are not.  I'm using no mods other than CQUI and it's the only version of CQUI present.  Rise & Fall with all DLCs.  First issue is that I don't get any info from tech or civic research, just says "ResearchName" with no time updates.  Second is that the City State list is limited to the first one I meet.  I can interact with the others and such, but I have to manually click on their geographical heading to do so.

![researchname](https://user-images.githubusercontent.com/381121/36797195-5ef28922-1c75-11e8-871d-2914243a04bc.png)

![citystates](https://user-images.githubusercontent.com/381121/36797166-4e292614-1c75-11e8-9752-92147213d5fc.png)

Let me know if you need the save file and I'll upload it.

**Comment by BlackSmokeDMax** [comment]
@radiation as far as I can tell, they haven't started working on compatibility with R&F yet. They are working on the base game patch only first.

**Comment by radiation** [comment]
@BlackSmokeDMax Oops, you're totally right.  Thanks!

**Comment by trafenrest** [comment]
[Logs.zip](https://github.com/Azurency/CQUI_Community-Edition/files/1768390/Logs.zip)

Okay, now I have the steam mod deactivated and really use CQUI from here.
In Loa.log it says
> cqui_main: CQUI Loaded! Update for v1.0.0.216 (beta)

Use the saved game I uploaded in one of my previous messages
1 Send IRENAEUS to the Holy Place near Kyoto
2 Create a religion
3 Close the message window
Soon, the audio loop will start triggered by two different events happen at the same time.
Now you'll need the task manager...

**Comment by rasputinoz** [comment]
got an issue with a "City Ranged Attack" the game wont let me proceed.

Save game attached.

I not sure where to find the log files you refer to in other posts. let me know and i will add them too.

Game had played will up until this point.


[MONTEZUMA 102 350 BC.zip](https://github.com/Azurency/CQUI_Community-Edition/files/1769056/MONTEZUMA.102.350.BC.zip)

**Comment by rasputinoz** [comment]
ok started a new game and i seem to have lost the city details refer to picture
![capture](https://user-images.githubusercontent.com/30841016/36820071-0d9bd3fe-1d3c-11e8-98a9-f622577266f2.JPG)

**Comment by rasputinoz** [comment]
ok seems like a strange problem, i uninstalled all mods. Then slowly readded them one by one and checked. Each time new mod added no problems, even when i added CQUI as last mod the problems have gone away.

**Comment by rasputinoz** [comment]
ok another issue strange one too. in current game the Romans are my ally but are also at war with me.

refer to this saved game 

i am in Roman territory and can attack on my turn despite the obvious alliance details

[MONTEZUMA 266 1775 AD.zip](https://github.com/Azurency/CQUI_Community-Edition/files/1769969/MONTEZUMA.266.1775.AD.zip)

**Comment by rasputinoz** [comment]
has anyone else had issues with Levying the military of a city state when you are its Suzerin.

I have a number of city states but cant levy any of them. The button is unclickable and the cost is highlighted as 0 Gold

**Comment by thesatch** [comment]
Hi,

I have 2 bugs with the last cqui version : 
1- About technology display at the top of the left, displaying "research name" without any information.
2- When I click on the cities states menu, I only have one city state and i can't add envoys to it.
This is a screenshot.
![image](https://user-images.githubusercontent.com/36992786/36903033-ae2518f8-1e2c-11e8-8122-53c5be77269d.png)

A part pf this, your mod is just awesome. ;)

**Comment by radiation** [comment]
@thesatch Those are Rise & Fall bugs.  This branch is only for updating the base game to the 1.0.0.216 patch; they haven't started on R&F yet.  I made the same mistake: https://github.com/Azurency/CQUI_Community-Edition/issues/283?_pjax=%23js-repo-pjax-container#issuecomment-369283571

**Comment by thesatch** [comment]
I installed rise and fall update a couple of days ago. CQUI didn't work so I decided to play without ... but it's difficult to go back ;).
So I tried the last cqui version and after severals tests, cqui started successfully. Just have this 2 bugs.

**Comment by the-gasman** [comment]
I've downloaded the latest patch from 2 days ago, and no there is no audio with pop-ups at all.

**Comment by ihendriks** [comment]
> I installed rise and fall update a couple of days ago. CQUI didn't work so I decided to play without ... but it's difficult to go back ;).
> So I tried the last cqui version and after severals tests, cqui started successfully. Just have this 2 bugs.

As @radiation tried to explain, CQUI does **not** work with the Rise & Fall DLC. It's still being tested for patch 1.0.0.216. When it works properly there, it'll be patched to work with the new DLC.

**Comment by wayneb64** [comment]
Playing with latest, no major issues so far, but the resources lens is missing. Seems like there might be others missing, but that one I use fairly often.

**Comment by thesatch** [comment]
@wayneb64 Don't you have the issue with the cities state menu ?
Regarding the lenses, most of the time it works for me. But sometimes I don't have green lens for farms.

**Comment by wayneb64** [comment]
My city states menu seemed fine. I certainly sent envoys and that worked. I didn't try to recruit their military or whatever that's called. I don't do that very often. I am talking about the lens called "Resources" that you can turn on/off luxuries, bonus resources and strategic resources individually or in groups. It's not that it's not working correctly, it's just that it's not there at all. It was normally near the end of the list.

**Comment by trafenrest** [comment]
Yep, I missed that too

**Comment by Azurency** [comment]
Yes, it was fixed in https://github.com/Azurency/CQUI_Community-Edition/pull/300 and it's merged now. There's also a new citizen overview lens. Thanks @astog 👍

I believe the mod will soon be finally updated on steam as there's not much issues left, and that we'll move on to fix it for Rise and Fall.

**Comment by MauiSven** [comment]
After the installation of [https://github.com/Azurency/CQUI_Community-Edition/tree/Patch-1.0.0.216](url) the trading lines and screens do not work anymore...

**Comment by Azurency** [comment]
@MauiSven I can't reproduce it, it's working fine for me. Do you have a save game and the logs please ?

**Comment by MauiSven** [comment]
@Azurency Here the savegame. But what "logs" do you mean?

[trade problem.zip](https://github.com/Azurency/CQUI_Community-Edition/files/1778509/trade.problem.zip)

**Comment by Azurency** [comment]
@MauiSven sorry, can't use that save... There are too much other mods. Can you try with the latest version I pushed today ? The logs is the `Logs` folder next to the `Saves` folder in your documents.

**Comment by MauiSven** [comment]
@Azurency Oh, I was afraid of that...
YESSSS, the actual version works correctly again regarding trade routes, thx!! 👍

**Comment by wayneb64** [comment]
I played some more with the latest as of Friday and the resource lens is back again, awesome! 

I noticed a few small issues: 
  - The city status report no longer remembers the sort you chose, it used to do that per load at least. The auto enable yields isn't enabling them on load like it should. 
 - The worker icons still get confused sometimes, much better than it used to be, but something shuts them off and you need to toggle a citizen and then they come back on for a while. 
 - There was something weird going on with city targeting. Not sure of the sequence but something along the lines of clicking the city targeting icon and then if you decide not to shoot something you can't select another unit and right clicking would not clear it either. The work around was to select a city and then pop back out again. 
 - I could not harvest citrus, but I have never tried that before and it's not explained anywhere in the rules. I could harvest a pasture however.

**Comment by TheHandiCraftsman** [comment]
> I could not harvest citrus, but I have never tried that before and it's not explained anywhere in the rules. I could harvest a pasture however.

Citrus is a luxury resource, pretty sure you can only harvest bonus resources.

> There was something weird going on with city targeting.

That issue has been on and off for me with vanilla no mods since release.  Maybe not an issue with CQUI.

**Comment by Azurency** [comment]
@wayneb64 I've tested the city attack range, and the right click is working for me, maybe you did the right click on the unit flag and that's why it did not work ?
Also left clicking your own unit is not doing anything and that's the excepted behavior because left clicking when in city attack mode means "attack the unit", so it does nothing when it's your own unit.

**Comment by wayneb64** [comment]
Every now and then I get locked into city attack mode and struggle to get out of it. Maybe user error, it just seemed a bit worse with latest, but that may have just been a fluke.

**Comment by Azurency** [comment]
@wayneb64 If you ever come across that issue again, can you post the logs files and a save game please :) It could be useful to reproduce the problem.

**Comment by lovejw2** [comment]
when i started my new game as Robert the Bruce the all the Research window said was 33 turns and ResearchName for both Science and Civic
[Logs.zip](https://github.com/Azurency/CQUI_Community-Edition/files/1783648/Logs.zip)

[AutoSave_0014.zip](https://github.com/Azurency/CQUI_Community-Edition/files/1783650/AutoSave_0014.zip)

**Comment by Azurency** [comment]
So... The new update 1.0.0.229 broke some things here and there in the base game CQUI. Here are the diffs https://github.com/Azurency/Civ6-UIFiles/commit/2b1d02a7b5575e9f921c68eb4e742c8b3fbc0fa3

**Files that need to be updated :**
- [ ] `actionpanel.lua`,
- [ ] `choosers/researchchooser.lua`,
- [ ] `civ6common.lua`,
- [ ] `civilopedia/civilopediasupport.lua`,
- [ ] `diplomacyactionview.lua`,
- [ ] `diplomacyactionview.xml`,
- [ ] `diplomacydealview.lua`,
- [ ] `diplomacyribbon.lua`,
- [ ] `instances/leadericon.xml`,
- [ ] `panels/citypanel.lua`,
- [ ] `panels/citypanel.xml`,
- [ ] `panels/citypaneloverview.lua`,
- [ ] `panels/citypaneloverview.xml`,
- [ ] `panels/modallenspanel.xml`,
- [ ] `panels/unitpanel.lua`,
- [ ] `panels/unitpanel.xml`,
- [ ] `partialscreens/citystates.lua`,
- [ ] `partialscreens/espionageoverview.lua`,
- [ ] `partialscreens/espionageoverview.xml`,
- [ ] `productionpanel.xml`,
- [ ] `screens/techtree.lua`,
- [ ] `supportfunctions.lua`,
- [ ] `worldtracker.lua`,
- [ ] `worldtracker.xml`,
- [ ] `worldview/citybannermanager.lua`,

**Comment by Azurency** [comment]
**update :** I just updated CQUI for the latest patch, I will try to start working on the Rise and Fall version this weekend.

**Comment by imagithubernow** [comment]
did 1.0.0.299 do something to prevent corps ? I was able to make a corps on a save this morning but cannot now.

**Comment by sparrow8332** [comment]
Steam now updated with latest master branch

**Comment by alexraileanu** [comment]
hi @Azurency. I'm using the master branch code on CIV version 1.0.0.229 and I'm experiencing a strange bug. I can't seem to open the Technology Tree at all. should I open a new issue for this?

**Comment by Azurency** [comment]
@alexeyOnGitHub ops, I didn't push the right version 🙈, fixed now.

**Comment by wayneb64** [comment]
So is the latest version on Steam now or should I still grab it from here?

Should I play offline to prevent 229 getting applied or are all the issues with 229 fixed in latest?

**Comment by Azurency** [comment]
@wayneb64 It's fixed on steam for 229, but it's not the latest version (I fixed an issue with future tech and the tech tree since)

**Comment by thyeestes** [comment]
I downloaded the latest version from the master branch, but still my units do not have this Circle above it.

**Comment by Azurency** [comment]
@thyeestes The circle on the map which shows health if the unit is damaged ? It should be shown correctly. Are you using any other mod ? The steam version should also work.

**Comment by wayneb64** [comment]
With latest applied to an existing game, I cannot form new Corp, no button for it shows up.

**Comment by imagithubernow** [comment]
@wayneb64 I don't think the corps problem is with cqui.  I started games without cqui (or any mods) at the modern era (which gives you nationalism) and there is no corps button.  Starting a game in the atomic era (when you get mobilization) you get a corps (and subsequent army) button.  I submitted a bug report to 2k.

**Comment by thyeestes** [comment]
@Azurency No I don't use any other mods at all. Apparently there was an old steam version in my workshop folder.Remove it and it works, I can see what kind of unit it is.

Thanks for this mod.

**Comment by Azurency** [comment]
I've made the diff for the Rise and Fall expansion https://github.com/Azurency/Civ6-UIFiles/commit/60ab84d04a53e89b3cbc365d77c15ed57fcefbdc beside the new additions in the `Additions` folder there are some changes in the `Replacements` folder, we'll have to look at each function there're extending to see if it conflict in any way with changes made by CQUI.

I'll try to make a list of file that needs to be looked at, but it will not be like previous updates as we'll also need to update the `modinfo` and create new files to keep the compatibility with the base game.

**Comment by wayneb64** [comment]
Naturalist length looks broken, marked tiles are several tiles off from wonder.

**Comment by alexraileanu** [comment]
@Azurency even with the latest master code (and civ version 1.0.0.229) I can not open the tech tree nor the culture tree. is this because of incompatibility with the latest version?

**Comment by alexeyOnGitHub** [comment]
I see overall CQUI has made a lot of progress in supporting RiseAndFall version.

with the latest Civ6 version + CQUI from Steam I can select techs and civics in their trees, but on-screen windows (top left corner) with tech and civic names (and their progress) are empty. they always show "Turns: 33" and "Research name" instead of actual names.

**Comment by alexeyOnGitHub** [comment]
two issues with "recommended city location":
* it is shown in an impossible location (close to a city)
* it is shown all the time, even though I did not even have a settler
![image](https://user-images.githubusercontent.com/239741/37567146-f086f89c-2a7f-11e8-8d46-af19ad0b875d.png)

**Comment by alexeyOnGitHub** [comment]
another issue with the latest Civ6 (v.229) with RiseAndFall + latest CQUI loaded from Steam:
cannot open City Details screen to see loyalty, buildings, etc. when I click on a city bar, it shows production panel on the right. but no city details on the left, where I expected to see loyalty tab and other city details.

**Comment by alexeyOnGitHub** [comment]
also, cannot send any envoys to a City State: clicking ">" arrow in envoys screen has no effect, and there is no "confirm" button

**Comment by alexeyOnGitHub** [comment]
anther one: "City States" panel does not show all cities I have met. it shows only one at the top.

**Comment by Azurency** [comment]
@alexeyOnGitHub CQUI is not compatible with Rise and Fall, there's no version to test yet

**Comment by Michaas** [comment]
The City States panel is bugged without Rise and Fall installed as well (v229). You can't see an overview of all the City States you've met, and you can't click on anything like ''Influenced by'' or ''Envoys sent''.

**Comment by alexeyOnGitHub** [comment]
> CQUI is not compatible with Rise and Fall, there's no version to test yet

sad panda
![image](https://user-images.githubusercontent.com/239741/37574036-147c1286-2adb-11e8-826d-b2c49b7fe3c9.png)

**Comment by Michaas** [comment]
Is there a way to load up an older version of civ6 like v220, because I don't recall having any problems back then.

**Comment by Azurency** [comment]
**small update** : I created the `Expansion1` branch and updated the citybanner for Rise and Fall

**Comment by Ketama303** [comment]
@Michaas 
Up until the v229 CQUI pretty much worked flawlessly for me. It was stupid to keep auto-update on. Now the whole Envoys tab is blank for me as well **without** the Rise & Fall update. 
And unless you have a system restore or drive backup image from prior to the patch it is not possible to roll-back the update.

**Comment by Sukayo** [comment]
How it looks, where help is needed?

**Comment by wayneb64** [comment]
Is there any status on work for Rise and Fall? It seems like it has been abandoned.

**Comment by ihendriks** [comment]
> Is there any status on work for Rise and Fall? It seems like it has been abandoned.
It's not abandoned, just not that many people that have time to work on it right now.

**Comment by Azurency** [comment]
Sorry, I was really busy this last few weeks. I just updated some more files for Rise and Fall on the `Expansion1` branch. I still need to work on the the production queue (to fix the Government Plaza issues), and some more details, but the `Expansion1` version should be more playable now.

**Comment by wayneb64** [comment]
No worries, I am in no real hurry for Rise and Fall, still playing Vanilla and enjoying CQUI. When it's ready for Rise and Fall I will get the upgrade.

**Comment by thejase** [comment]
@azurency we all appreciate all the hard work you and others are putting into this, especially with busy lives. Any way we can buy you a beer (donate) in thanks?

**Comment by deiga** [comment]
@wayneb64 If you're in no hurry then why would you use such strong language (abandoned)? 
It seems a bit disrespectful towards all the people who volunteer their free time to build something for us for free.

**Comment by wayneb64** [comment]
I meant no disrespect, I just hadn't seen any status for what seemed like a long time so I feared interest had gone away.

**Comment by CMFIC** [comment]
@Azurency I prefer waiting on your progress than waiting on any update from Firaxis/Take2

**Comment by poundjd** [comment]
I usually don't (now!!) even ask if thing are OK, for at least six months!  After a Year I ask if they are coming back and if no responses after two more years, I look to see if I can find someone who is better at this whole business to replicate the effects....
-jeff

**Comment by rasputinoz** [comment]
ok I don't know if it is relevant or just a coincidence. Been playing Single player Rise anf Fall with the Expansion version and have encountered the usual bugs ,that is Research and Civics not showing what is currently being researched and the City States Envoy screen not working. For some unknown reason I started a Multiplayer game of Rise and Fall on Internet mode. I started game with no other players and it wasn't until my first envoy popped up I noticed that it was working. In great delight I then noticed the research and civic boxes were working too. Strange I thought. 
I then started a Single player game. And lo and behold it was still working.

Could the two events be linked? 

Will try a few more games with different settings and maps and see what happens.

**Comment by r0h1t4sh** [comment]
+1 I have also encountered the issue where city states envoys screen
doesn't work

**Comment by rasputinoz** [comment]
Still no issues with the BETA version. Well done guys , appears to be working well now.

**Comment by wayneb64** [comment]
Where is this BETA version and what exactly is it? Sorry for my GitHub ignorance but I get completely lost clicking around in GitHub. Is this BETA version the latest version of code for Rise and Fall?

**Comment by rasputinoz** [comment]
Under the code section , 3 branches , choose the Expansion . Not the master one

Sent from my iPhone

On 21 Apr 2018, at 10:33 pm, wayneb64 <notifications@github.com<mailto:notifications@github.com>> wrote:


Where is this BETA version and what exactly is it? Sorry for my GitHub ignorance but I get completely lost clicking around in GitHub. Is this BETA version the latest version of code for Rise and Fall?

**Comment by oro1989** [comment]
@rasputinoz does the Expansion branch work with Rise and Fall and 1.0.0.216, or 1.0.0.229 ???

**Comment by Nijin22** [comment]
@oro1989 Not completely. I managed to get the game running, but some features are still not working (like sending envoy to city states - see #321)

**Comment by YangMeng0721** [comment]
Right, I had the same tech/civic not showing and city-state envoys issues before. When I checked out the `expansion1` branch, those problem had been solved. @Nijin22 Are you sure you are on branch `expansion1`?

**Comment by Nijin22** [comment]
@YangMeng0721 - yes, I just double checked. I'm on 3df8664

**Comment by duvaljd** [comment]
Thanks @Azurency for all your work here! Love this mod, can't play Civ VI without it anymore.

I'm using the expansion1 branch on Rise & Fall. I can't seem to figure out how to see Loyalty information - other than the Lens, which only lets you see where the sources of Loyalty are coming from, not specific numbers. Is anyone else experiencing this bug?

**Comment by ajbrown** [comment]
I'd suggest either keeping a list of "actual things that are broken" in the issue, or making sure there's an issue linked to this one for every actually thing that is broken.  That'll help people understand what bugs the should and shouldn't be reported, as well as a good indication of progress. I love this mod and have some PM and dev skills, so wouldn't mind helping out in that regard if needed. 

Thanks for all of the hard work!

**Comment by ihendriks** [comment]
> I'd suggest either keeping a list of "actual things that are broken" in the issue, or making sure there's an issue linked to this one for every actually thing that is broken. That'll help people understand what bugs the should and shouldn't be reported, as well as a good indication of progress. I love this mod and have some PM and dev skills, so wouldn't mind helping out in that regard if needed.

Yeah I agree there's basically no overview about the state of the project. I've been following pretty closely since the release of R&F and I don't even know where we stand at this point. Don't have the time to really figure out what needs to be looked at either, so I end up not doing anything at all... I'm sure others have the same problem in that regard.

Maybe it's an idea to get some more people into some sort of core-maintainer role to remove some weight from @Azurency as well. Maybe we could discuss it on Discord with everyone interested in helping maintain the mod?

**Comment by Azurency** [comment]
## Update on CQUI for Rise and Fall

### How anyone can help
I updated some more files, it's time to **start testing** the `Expansion1` branch 🎉. Be sure when doing so to install it cleanly (removing old version, workshop version, etc.).
Some people reported that the city state screen did not allow them to send envoys, so it's something to look at.

⚠️ When you have found a bug : 

- post a new message with **the logs and a save game** (in your `document/my games/sid Meier's .../` you have a logs folder and the save game are in saves)

- do not take the time to post a bug that occurred while you had other mods enabled, only use cqui when testing

### How developers can help
There's still some work to be done : 
- [ ] look at the `URS/Expansion1/Replacements/ReportScreen_Expansion1.lua` file, there's still a bug with the city status report tab
- [ ] look at the `Government Screen` you can find [the new file created by Firaxis](https://github.com/Azurency/Civ6-UIFiles/blob/expansion1/Replacements/GovernmentScreen_Expansion1.lua) for the expansion and figure out why the _My Government_ tab is showing (a feature of CQUI is that this tab is hidden), then create a corresponding file in cqui (under `/Expansion1/Replacement`).

**Comment by ihendriks** [comment]
@Azurency 
> look at the Government Screen you can find the new file created by Firaxis for the expansion and figure out why the My Government tab is showing (a feature of CQUI is that this tab is hidden), then create a corresponding file in cqui (under /Expansion1/Replacement).

The new file shouldn't have any affect on the removal of that tab, but I can't figure out where it's going wrong either.
Makes more sense that there's an issue with the mod's governmentscreen.lua, since they slightly changed the window/screen size calculation and maybe that's breaking it. But I didn't get that far.

**Comment by rasputinoz** [comment]
Thanks for the latest update. Installed and testing. No issues after starting three random games.

**Comment by PRoosta** [comment]
@Azurency 

So I just tested the latest CQUI Expansion 1 version on 1.0.0.216 and the Tech Tree does not work/show up at all. The Civic Tree works just fine unlike what is mentioned in #326. 

I've included the [Logs](https://github.com/Azurency/CQUI_Community-Edition/files/1958950/Logs.zip) and even a [save file](https://github.com/Azurency/CQUI_Community-Edition/files/1958951/PETER.1.4000.BC.zip) but this should be reproducible by starting any singleplayer game as far as I can tell.

**Comment by Azurency** [comment]
@PRoosta Yes, you should be on the latest version. I just updated the testing instructions to make that clear to everyone.

**Comment by BlackSmokeDMax** [comment]
So far, mostly so smooth.

Only issue I have seen is with auto-applying the scout lens, it is just not happening. I can select it via the lens button near the mini map and it works just fine. Just not auto turning on while scout is selected.

I have checked in the CQUI options and it *is* turned on.

This game is with the Cree, who happen to have the special scout unit, the Okihtcitaw. Perhaps that is the reason. Think I will start a game with a civ with a basic scout unit and see what happens.

Version, is 1.0.0.229, with R&F.

**Comment by camk16** [comment]
Waterpark pin is bugged?

![untitled](https://user-images.githubusercontent.com/30790283/39506311-926b0a20-4d8c-11e8-9785-7abde6a0abd2.png)

**Comment by mattflynn-dev** [comment]
Found a very, very minor bug.
Changing the map overlay options to always show tile yields doesn't carry over in the save state. Upon each load, I have to select to show the yield overlay manually. The base game saves remember this on load.
This isn't significant issue by any means, but thought it would be useful to report in case it has knock-on effects elsewhere.

[Logs.zip](https://github.com/Azurency/CQUI_Community-Edition/files/1969186/Logs.zip)

[quicksave.zip](https://github.com/Azurency/CQUI_Community-Edition/files/1969189/quicksave.zip)

**Comment by alexeyOnGitHub** [comment]
I downloaded the latest version of "expansion" branch, tried it with Civ6 RaF 1.0.0.229 (build 339082). I still have  this https://github.com/Azurency/CQUI_Community-Edition/issues/326 issue.
also, cannot send envoys to city states.
note - I have the old "official" CQUI version (from Steam workshop) listed in "additional content" alongside with the one I downloaded, but I disabled that one. not sure if this is relevant.

**Comment by Nansen17** [comment]
I have played ca. 150 turns with the Spring 2018 update and your Expansion1 test build without any problems. All features I tried were working properly. Game was with Barbarossa, and the only other mod active was Detailed Worlds (which as far as I understand only influences map generation).
Great work, keep it up!

**Comment by ihendriks** [comment]
@Nansen17 Did you try the new trade option to join an ongoing war (or ask AI to join)?

**Comment by Nansen17** [comment]
No, I didn't, but will post again once I had a chance to do so.

**Comment by Nansen17** [comment]
@ihendriks: I have tried it now, and I can see the menu options for both parties under "deals", but whichever details I chose there is a exclamation mark in red with "Offer blocking deal" and I can't close a deal. Will have to start a new game without CQUI to see if it works without.

**Comment by Nansen17** [comment]
@ihendriks 
In a regular game without CQUI the options to join a war are working just fine, so there is still a minor issue remaining there ...

**Comment by ihendriks** [comment]
@Nansen17 Thanks for testing this. Just to check: with CQUI you can see the options on the trade screen, they just don't ever seem to work. Is that correct?

**Comment by Nansen17** [comment]
@ihendriks Hi Ivo, that is correct. You can see the options with CQUI and even select a war to join, only the "Accept" button doesn't work, being blocked by the exclamation mark in a red diamond next to the "joint war" item on the trade list. 
This seems to be a blocking mechanism used by Civ6 on other occasions too. I have seen it in trade negotiations with other empires when they are asked to include trade items they cannot give up.

**Comment by ihendriks** [comment]
@Nansen17 Yeah the AI doesn't want to trade certain things (Usually cities and non-duplicate luxuries), so it might just be that it's the AI unwilling to trade the joint-war.

Might also be that since our R&F version isn't quite ready yet, that it's broken on our end. I'll look into this when our Spring update is released and we continue working on R&F.

**Comment by Nansen17** [comment]
@ihendriks Hi Ivo, that is correct. You can see the options with CQUI and even select a war to join, only the "Accept" button doesn't work, being blocked by the exclamation mark in a red diamond next to the "joint war" item on the trade list. 
This seems to be a blocking mechanism used by Civ6 on other occasions too. I have seen it in trade negotiations with other empires when they are asked to include trade items they cannot give up.

**Comment by ambulanced** [comment]
hey boys... just wanted to drop in and say I love you all.  I hope you all update this mod when you can, it was amazing. the production  queue, the scout and builder tile colors, the camera movement, and trade and HUD info!

This mod made the games with my friends so much faster.  Now we have to wait for one person because they have 12 cities with no queue.  

Don't give up! You can do it!

**Comment by kecker79** [comment]
So has this effort been abandoned?

**Comment by Ketama303** [comment]
@kecker79
As far as I'm aware of, no.  We're all eager to have a working version again, but he does this in his free time, so please be patient.

**Comment by ihendriks** [comment]
@kecker79 It's not abandoned, but the maintainer is currently stuck with his job and there's noone else able to push changes. When he's back it should be a quick release.

**Comment by ComanDanteDevOps** [comment]
I am but a lowly DevOps Engineer but if I can assist with getting this out there quicker I am fairly capable in GitHub at least.  I'm playing with the latest expansion1 build with the fix for Tech Tree and fix for the Joint War/Diplomacy issues as well.  As far as I can tell there are no major bugs I am seeing.  I can see Pull Requests for these same as well.  So looks like it would just take a review & merge to get at least a functional build out.  I could manage that if the help would be welcome.

**Comment by HicPuer** [comment]
Where I can find fix for the Joint War/Diplomacy issue?
Is the "fix" this: "Removing the "IDS" folder from integrations...." https://github.com/Azurency/CQUI_Community-Edition/issues/340#issuecomment-394294535

**Comment by ihendriks** [comment]
> @ComanDanteDevOps
> I am but a lowly DevOps Engineer but if I can assist with getting this out there quicker I am fairly capable in GitHub at least. I'm playing with the latest expansion1 build with the fix for Tech Tree and fix for the Joint War/Diplomacy issues as well. As far as I can tell there are no major bugs I am seeing. I can see Pull Requests for these same as well. So looks like it would just take a review & merge to get at least a functional build out. I could manage that if the help would be welcome.

There's only a single maintainer, as long as he's not here nothing can be merged or pushed to steam.

**Comment by ihendriks** [comment]
> @HicPuer 
> Where I can find fix for the Joint War/Diplomacy issue?
> Is the "fix" this: "Removing the "IDS" folder from integrations...." #340 (comment)
Removing the IDS folder will rever the diplomacy view to the default, meaning joint wars etc. will indeed work. It doesn't fix the issue, it just removes the functionality altogether.

> @HicPuer 
> whitenoise500, unoffical fix for report screen is to delete "CQUI_Community-Edition-expansion1\Integrations\URS" and replace this with latest version: forums.civfanatics.com/resources/unit-report-screen.25396
We've actually been missing this one, thanks for that!

**Comment by ComanDanteDevOps** [comment]
There is a proper fix for the diplomacy view as well:
https://github.com/Azurency/CQUI_Community-Edition/pull/346

I may take a stab at forking and putting these various fixes together.  Not to replace this version but just to at least have something we can use and test against until we hear back from @Azurency .  Longer term, I'd be happy to help with maintaining if it's welcome.

**Comment by kecker79** [comment]
@ComanDanteDevOps I think both a fork and your maintenance efforts would be greatly appreciated.

**Comment by ambulanced** [comment]
is it ironic that @Azurency has become Chao?

**Comment by jeffpierce** [comment]
Not really.  Stuff happens, whether that’s being busy or getting bored.

**Comment by comandante-cinc** [comment]
I hope @Azurency is ok.  Am appreciative of the efforts he put into this mod.  At this point though it's been a long time since we've heard from him.  I hope he's ok and doing well but in the meantime, I have created a fork and merged in all of the fixes I have seen here.  It should essentially be the same as the package @HicPuer put together.

I am willing to maintain the mod but I am not a developer.  If you are able to contribute to fixes or even new features, I'm sure the community would be happy to see them.  I can definitely do the work of reviewing, merging, testing etc.  I will maintain this as a fork, and should Azurency return, it would be a quick task to merge it back in.

Fork is here:
https://github.com/ComanDanteDevOps/CQUI_Community-Edition

**Comment by Azurency** [comment]
I'm ok, somehow much more busy that I would like... I'm sorry for being away so long, haven't played civ in ages 😭. 
I'm going to take a look at everything this weekend, hopefully get everything working (the main problem was the `diplomacydealview.lua`, I'll review the pull request #346). If I don't have enough time this weekend, I'll be free for sure at the end of next week to work on that.
_As I'm not owning the steam repository, it may take a little more time for Sparrow to upload an updated version._

**Comment by sparrow8332** [comment]
@comandante-cinc  @Azurency is very busy atm with real life stuff and myself also. I was planning to have some free time last week but have family stuff going on atm. I dont know how long @Azurency is afk for but be patient he will return, and i will also once real life gives a break to do so.  The mod hasnt been forgotten about or dropped just on pause atm.

**Comment by ComanDanteDevOps** [comment]
My apologies if I came off as impatient.  Just wanted to have the known fixes together so some could play and test.  It seems to be working great with those three fixes applied for RnF at least.  I don't see any major bugs.  I'd love to assist in maintaining CQUI in the future by testing, reviewing and merging as needed.  I could definitely make your work easier by handling some of the DevOps type tasks.

**Comment by ambulanced** [comment]
OMG guys it's Monday! Are there new gifts under the tree?

**Comment by Azurency** [comment]
Ok, so the `expansion1` branch have been upgraded, the diplomacydealview should now work and I made a quick fix for the report screen (not perfect at all, there's some overflow in the city status tab).
Everything should be 100% functional now.

**Comment by Azurency** [comment]
**Update** : Everything is merged into master, the steam version will soon be updated. Thank you all for your help and your patience 👍.
I'm going to test it on Mac now.

**Comment by ncl2fth** [comment]
Thanks everyone, I’m sure I can speak for all of us that don’t have the knowledge to help, your work is appreciated.

**Comment by pelff** [comment]
Thx a lot for all the work done! This mod is just awesome! Thx again.

**Comment by ambulanced** [comment]
The gods have forgiven us.

**Comment by Azurency** [comment]
Everything is now updated on Steam, time to close this issue at last.
